core/src/main/java/org/frankframework/util/XmlUtils.java

```diff
@@ -1302,25 +1303,32 @@ public static void setTransformerParameters(Transformer t, Map<String, Object> p
 		if (parameters == null) {
 			return;
 		}
-		for (String paramName:parameters.keySet()) {
-			Object value = parameters.get(paramName);
+		for (Map.Entry<String, Object> entry : parameters.entrySet()) {
+			String paramName = entry.getKey();
+			Object value = sanitizeValue(paramName, entry.getValue());
 			if (value != null) {
-				if (value instanceof Reader || value instanceof InputStream || value instanceof byte[] || value instanceof Message) {
-					try {
-						value = MessageUtils.asString(value);
-					} catch (IOException e) {
-						throw new IOException("Cannot get value of parameter ["+paramName+"]", e);
-					}
-				}
 				t.setParameter(paramName, value);
 				log.debug("setting parameter [{}] on transformer from class [{}]", paramName, value.getClass().getTypeName());
-			}
-			else {
+			} else {
 				log.info("omitting setting of parameter [{}] on transformer, as it has a null-value", paramName);
 			}
 		}
 	}
 
+	private static Object sanitizeValue(String paramName, Object value) throws IOException {
+		if (value == null) {
+			return null;
+		}
+		if (value instanceof Reader || value instanceof InputStream || value instanceof byte[] || value instanceof Message) {
+			try {
+				return MessageUtils.asString(value);
```

Comment:
Die methode is deprecated, is daar een ander alternatief voor?

Reply:
Zelf als message wrappen, maar dat vind ik misbruik van het Message object. Ik weet dat de MessageUtils.asString() dat nu ook doet, maar ik "verberg" dat liever in die method zodat we het probleem uiteindelijk 1x daar oplossen.